app/src/main/java/com/zulip/android/activities/LoginActivity.java

```diff
@@ -265,6 +182,93 @@ public void onTaskFailure(String result) {
         }
     }
 
+    private void setupSignIN(String clientId) {
+        AuthorizationServiceConfiguration serviceConfiguration = new AuthorizationServiceConfiguration(
+                Uri.parse("https://accounts.google.com/o/oauth2/v2/auth") /* auth endpoint */,
+                Uri.parse("https://www.googleapis.com/oauth2/v4/token") /* token endpoint */
+        );
+
+        Uri redirectUri = Uri.parse("com.zulip.android:/oauth2callback");
+        AuthorizationRequest.Builder builder = new AuthorizationRequest.Builder(
+                serviceConfiguration,
+                clientId,
+                AuthorizationRequest.RESPONSE_TYPE_CODE,
+                redirectUri
+        );
+        builder.setScopes("profile", "email");
+        AuthorizationRequest request = builder.build();
+        AuthorizationService authorizationService = new AuthorizationService(LoginActivity.this);
+        String action = "com.zulip.android.HANDLE_AUTHORIZATION_RESPONSE";
```

Comment:
This is probably a good thing to keep as a constant, rather than inline as well.